interactive/app_utils.py

```diff
@@ -0,0 +1,214 @@
+            s_img["label"] for s_img in IMAGE_EXAMPLES[st.session_state.category_option]
+        ]  # get captions
+        st.session_state.example_imgs = [
+            Image.open(s_img["path"])
+            for s_img in IMAGE_EXAMPLES[st.session_state.category_option]
+        ]  # get images
+        example_images_zip = dict(
+            zip(st.session_state.example_captions, st.session_state.example_imgs)
+        )
+        img_selection = st.selectbox(
+            f"Choose an image - {st.session_state.category_option}.",
+            example_images_zip,
+        )  # select image
+        st.session_state.selected_img = example_images_zip[img_selection]
+        st.session_state.show_input_img = True
+
+    def create_image_provision_for_random_storage_pull(self):
+        """
+        Resets state of previous provisioning selection and pulls from local/cloud storage a category-specific list of
+        image examples that a user can select from the list. Additionally, a button for re-running random selection is
+        implemented together with the input option for the number of sampled images.
+        """
+        self.states_manager.reset_states_after_image_provisioning_list()
+        st.session_state.pull_random_img_number = st.number_input(
+            label=f"Choose random images from {st.session_state.category_option} category.",
+            value=5,
+            min_value=1,
+            format="%i",
+        )
+        if st.button("Generate Images"):
+            (
+                st.session_state.random_captions,
+                st.session_state.random_imgs,
+            ) = env_handler.get_random_images_from_collection(
+                collection_name=st.session_state.category_option,
+                k=st.session_state.pull_random_img_number,
+            )  # Pulls a sampled set of images from local/cloud storage
+        if (
+            st.session_state.random_captions is not None
+            and st.session_state.random_imgs is not None
+        ):
+            random_images_zip = dict(
+                zip(
+                    st.session_state.random_captions,
+                    st.session_state.random_imgs,
+                )
+            )
+            img_selection = st.selectbox("Choose an image", random_images_zip)
+            st.session_state.selected_img = random_images_zip[
+                img_selection
+            ]  # returns an image based on selection
+            st.session_state.show_input_img = True
+
+    def create_image_provision_for_manual_upload(self):
+        """
+        Resets state of previous provisioning selection and provides upload button for a user.
+        Any RGB image can be uploaded.
+        """
+        self.states_manager.reset_states_after_image_provisioning_list()
+        st.markdown(f'<p class="big-font">Choose a file.</p>', unsafe_allow_html=True)
+        byte_img = st.file_uploader("Upload an image from a local disk.")
+        if byte_img is not None:
```

Comment:
check if `if byt_img` is working as well

Reply:
Done